### DISCUSSION TITLE
WaveOutEvent only playes a single time.

### DISCUSSION THREAD

**Discussion by DasLixou** [naudio/NAudio#915] [Q&A]
Here I have my code:

```csharp
internal class PlayCommand {
  private WaveOutEvent outputDevice;
  private AudioFileReader audioFile;
  
  public PlayCommand()
  {
      outputDevice = new WaveOutEvent();
  
      audioFile = new AudioFileReader("Assets/Samples/lixoubass.wav");
      outputDevice.Init(audioFile);
  }
  
  public bool CanExecute(object? parameter) => true;
  
  public void Execute(object? parameter)
  {
      outputDevice.Stop();
      outputDevice.Play();
  }
}
```

but it only plays once. after I executed it once, it couldn't get played a second time.

I also modified the code to match with the wiki like so

```csharp
internal class PlayCommand {
  private WaveOutEvent outputDevice;
  private AudioFileReader audioFile;

  private void OnPlaybackStopped(object sender, StoppedEventArgs args)
  {
      outputDevice.Dispose();
      outputDevice = null;
      audioFile.Dispose();
      audioFile = null;
  }
  
  public bool CanExecute(object? parameter) => true;
  
  public void Execute(object? parameter)
  {
      outputDevice?.Stop();

      outputDevice = new WaveOutEvent();
      outputDevice.PlaybackStopped += OnPlaybackStopped;

      audioFile = new AudioFileReader("Assets/Samples/lixoubass.wav");
      outputDevice.Init(audioFile);

      outputDevice.Play();
  }
}
```

Here it works and playes multiple times, but when i play it while it was playing, it stops with a cracky sound and dont play for the second time. Anyone know how to fix this?

Thanks :D

**Comment by smay4finger** [comment]
If you wanna play the file again, you have to rewind it with a Seek(0). The crack you hear might be caused by a large difference of the output sample. For example if your audio wave ends with a sample of +1.0 and starts with -1.0 you have a huge change that results in a steep edge.

You might want to edit your lixoubass.wav to include an envelope with attack and decay.

Hope this helps, although answer comes late.

**Reply by DasLixou** [reply]
Ok I’ll try it with Seek(0). But the crack shouldnt be by the audio file because I tried it in vlc groovy audio and audacity and there is sounds ok
